### PR TITLE
Add tests for nil *Resource

### PR DESCRIPTION
**Pull request by jmacd**
There is a precedent in `sdk/resource` for allowing a `nil` resource value. One method did not have such a test, and this adds it (along with some tests).

### CODE REVIEW

**Review by Aneurysm9 — APPROVED** [review]

File: sdk/resource/resource_test.go [hunk 1]
```diff
@@ -205,6 +205,17 @@ func TestMerge(t *testing.T) {
 	}
 }
 
+func TestEmpty(t *testing.T) {
+	var res *resource.Resource
+	require.Equal(t, "", res.SchemaURL())
```
Comment (pellared) [inline sdk/resource/resource_test.go:210]: I believe that you can use `assert` instead of `require` here and in the following lines

**Review by XSAM — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

File: sdk/resource/resource_test.go (hunk 1 above)
Comment (jmacd) [reply sdk/resource/resource_test.go]: Done.

**Review by MrAlias — APPROVED** [review]
